### ISSUE TITLE
Plugins should be able to provide a synposis to a log file

### ISSUE THREAD

**Issue by Kittyfisto** [Kittyfisto/Tailviewer#207]
Whenever I'm debugging issues in one application or the other, I find myself constantly changing filters and scrolling over the entire log file to find some arcane information such as:

- Application version
- Environment variables
- Length of log in minutes / hours / days
- Even more application specific information

Tailviewer cannot do very much on a general front (besides finishing the goddamn analysis feature), however it could provide a plugin interface which allows the following:

## Define a custom control which is displayed *next to* the log file itself
Tailviewer could, for example, provide another tab on the right side panel bar which expands when clicked on. This panel could then display the custom control provided by the log file. An MSBuild plugin might for example display the build time as well as a per-project status.

## Allow user interaction between the synposis tab and the log file itself
A plugin could provide certain controls which filter the log file in a certain way which is *always* necessary for that type of log file. So for example a plugin for an MSBuild log could provide easy to reach filters which filter the output to display only the log entries pertaining to a particular project.

## Proposed interface
```
public interface ILogFileSynopsisPlugin
    : IPlugin
{
    ILogFileSynopsisViewModel CreateViewModel(IServiceContainer services, ILogFile logFile);
    FrameworkElement CreateControl(IServiceContainer services, ILogFileSynopsisViewModel viewModel);
}
```

**Comment by Kittyfisto** [comment]
Done. Will be part of the 0.9 release.

**Comment by Kittyfisto** [comment]
Fixed with [v0.9.0](https://github.com/Kittyfisto/Tailviewer/releases/tag/v0.9.0)
